### DISCUSSION TITLE
Adding custom operations  on pip install <my_package>[<extra>]

### DISCUSSION THREAD

**Discussion by dboeckenhoff** [python-poetry/poetry#7869] [Q&A] [answered]
When the user installs my package via `pip install <my_package>[triggering_extra]`, specifying the extra 'triggering_extra' I want to take some action (by calling some script / function).

I have looked into the docu and code and have not seen how to do that.
Is it possible at all?

Because setup.py is not called any more, I don't know, where even to put a workaround, e.g. re.match sys.argv for [triggering_extra].
Are there workarounds ?

**Comment by finswimmer** [comment, marked as answer]
Hey @dboeckenhoff,

"post install" script are discourage because they are usually unexpected and can be dangerous. So this is not supported by Poetry.

Instead your package should provide a [command](https://python-poetry.org/docs/pyproject/#scripts) that must be run by the user explicit after installing.

fin swimmer

**Reply by dboeckenhoff** [reply]
Thanks. That makes sense. Interestingly, setup.py does allow such things with setuptools. Following your answer wouldn't that leave a vulnerability in all of the setuptools universe?

**Comment by dboeckenhoff** [comment]
If I use my_package as a dependency in another library, how would I trigger the script without cloning the source?
